### PR TITLE
Truncate GCE load balancer names to 63 chars

### LINKED ISSUE #6812: Limit length of load balancer names
GCE has a length limit of 63 characters for its load balancer components (target pools and forwarding rules), and AWS has a length limit of 32 characters. Our current load balancer name construction of clustername-namespace-servicename can exceed both limits, especially AWS's. We should use a shorter name construction method less likely to hit these limits.

### PR DESCRIPTION
**Pull request by brendandburns**
@a-robinson 

Closes #6812

### CODE REVIEW

File: pkg/cloudprovider/gce/gce.go [hunk 1]
```diff
@@ -266,10 +269,23 @@ func translateAffinityType(affinityType api.AffinityType) GCEAffinityType {
...
 
+func normalizeName(name string) string {
+	// If it's short enough, just leave it.
+	if len(name) < LOAD_BALANCER_NAME_MAX_LENGTH-6 {
+		return name
+	}
+	// Hash and truncate
+	hash := md5.Sum([]byte(name))
+	truncated := name[0 : LOAD_BALANCER_NAME_MAX_LENGTH-6]
+	shortHash := hash[0:6]
```
Comment (roberthbailey) [inline pkg/cloudprovider/gce/gce.go]: nit: can just do [:6] here

**Comment by a-robinson** [comment]
LGTM

**Comment by caesarxuchao** [comment]
Maybe I've missed something. The change makes sense to me, but are you going to revert the changes in #7145? If not, the origName will always be shorter than 32 bytes because of the logic in GetLoadBalancerName(), so the hash function will never be called.

**Comment by quinton-hoole** [comment]
I'd propose rolling this PR back until we've thought this through and tested it better.  
I've had to roll back #7635 which attempted to roll back #7145, the former because it simply didn't work, and broke e2e tests.
Which means that this PR actually just introduces dead code, as per @caesarxuchao's comment above.
I also have some other concerns that I'd like to bring up in the review of the reformulation of this PR, on a week day :-)

**Pull request merged**